templates/terraform/custom_flatten/secret_version_access.go.erb

```diff
@@ -0,0 +1,43 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2020 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+func flatten<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d *schema.ResourceData, config *Config) interface{} {
+	transformed := make(map[string]interface{})
+
+	url, err := replaceVars(d, config, "{{SecretManagerBasePath}}{{name}}:access")
+	if err != nil {
+		return err
+	}
+
+	parts := strings.Split(d.Get("name").(string), "/")
+	project := parts[1]
+
+	accessRes := make(map[string]interface{})
+	accessRes, err = sendRequest(config, "GET", project, url, nil)
+	if err != nil {
+		// if this secret version is disabled, the api will return an error, as the value cannot be accessed
+		if strings.Contains(err.Error(), "is not in ENABLED state") && d.Get("enabled").(bool) == false {
```

Comment:
What would happen if the API error message changed at some point in the future? Trying to read or do any operations on a disabled secret version would throw an error, right?

Is there a way we can check the state of `enabled` and if it is disabled, not even make this call?

Though it might entail depending on the order of the calls of the flatten functions, and I'm not sure that's a change we want to make

Reply:
Yes, that's correct.  The error bubbles up to the `flattenedProp` loop, which is less than ideal because it panics when it tries to cast it into a string, but yea - I suppose I can add a check in there to see if it's an error so that it errors nicely in that case.
I tried checking the state of enabled, but yea, it was an order deal, I think.  I'll run through again and see specifically what case that was in.  I did a lot of testing/moving around code with this one, so it could be that after I fixed something else, this might work now.

Reply:
I would try to get this flatten call to happen after we flatten the state so that we can check `d.Get("state")` and not make the `:access` call if state is disabled. Maybe moving the fields around within `api.yaml` can fix it?

Reply:
That worked!  Thanks!